tasks/static_quality_gates/lib/gates_lib.py

```diff
@@ -136,6 +136,31 @@ def _add_gauge(self, timestamp, common_tags, gate, metric_name, metric_key):
             )
         return None
 
+    def generate_relative_size(self, ctx, filename="static_gate_report.json", ancestor=None):
+        if ancestor:
+            out = ctx.run(
+                f"aws s3 cp --only-show-errors --region us-east-1 --sse AES256 s3://dd-ci-artefacts-build-stable/datadog-agent/static_quality_gates/{ancestor}/{filename} {filename}",
+                hide="stdout",
+                warn=True,
+            )
+            if out.exited == 0:
+                ancestor_metric_handler = GateMetricHandler(ancestor, self.bucket_branch, filename)
+                for gate in self.metrics:
+                    ancestor_gate = ancestor_metric_handler.metrics.get(gate)
+                    if not ancestor_gate:
+                        continue
+                    for metric_key in ["current_on_wire_size", "current_on_disk_size"]:
+                        if self.metrics.get(metric_key) and ancestor_gate.get(metric_key):
+                            relative_metric_size = ancestor_gate[metric_key] - self.metrics[gate][metric_key]
+                            self.register_metric(gate, metric_key.replace("current", "relative"), relative_metric_size)
+            else:
+                print(
+                    color_message(
+                        f"[WARN] Unable to fetch quality gates {filename} from {ancestor} !",
```

Comment:
I think it would be better to include the error logs to ease debugging in case this happens